packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
+    return isInAccessibilityMode != oldWidget.isInAccessibilityMode;
+  }
+}
 
-  /// The widget below this widget in the tree.
-  ///
-  /// Typically a [Text] widget.
-  final Widget child;
+// iOS style layout policy widget for sizing an alert dialog's content section and
+// action button section.
+//
+// See [_RenderCupertinoDialog] for specific layout policy details.
+class _CupertinoDialogRenderWidget extends RenderObjectWidget {
+  const _CupertinoDialogRenderWidget({
+    Key key,
+    @required this.contentSection,
+    @required this.actionsSection,
+  }) : super(key: key);
 
-  /// Whether the button is enabled or disabled. Buttons are disabled by
-  /// default. To enable a button, set its [onPressed] property to a non-null
-  /// value.
-  bool get enabled => onPressed != null;
+  final Widget contentSection;
+  final Widget actionsSection;
 
   @override
-  Widget build(BuildContext context) {
-    TextStyle style = _kCupertinoDialogActionStyle;
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoDialog(
+      dividerThickness: _kDividerThickness / MediaQuery.of(context).devicePixelRatio,
+      isInAccessibilityMode: _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false,
+    );
+  }
 
-    if (isDefaultAction) {
-      style = style.copyWith(fontWeight: FontWeight.w600);
+  @override
+  void updateRenderObject(BuildContext context, _RenderCupertinoDialog renderObject) {
+    renderObject.isInAccessibilityMode = _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false;
+  }
+
+  @override
+  RenderObjectElement createElement() {
+    return _CupertinoDialogRenderElement(this);
+  }
+}
+
+class _CupertinoDialogRenderElement extends RenderObjectElement {
+  _CupertinoDialogRenderElement(_CupertinoDialogRenderWidget widget) : super(widget);
+
+  Element _contentElement;
+  Element _actionsElement;
+
+  @override
+  _CupertinoDialogRenderWidget get widget => super.widget;
+
+  @override
+  _RenderCupertinoDialog get renderObject => super.renderObject;
+
+  @override
+  void visitChildren(ElementVisitor visitor) {
+    if (_contentElement != null) {
+      visitor(_contentElement);
     }
+    if (_actionsElement != null) {
+      visitor(_actionsElement);
+    }
+  }
 
-    if (isDestructiveAction) {
-      style = style.copyWith(color: CupertinoColors.destructiveRed);
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    _contentElement = updateChild(_contentElement, widget.contentSection, _AlertDialogSections.contentSection);
+    _actionsElement = updateChild(_actionsElement, widget.actionsSection, _AlertDialogSections.actionsSection);
+  }
+
+  @override
+  void insertChildRenderObject(RenderObject child, _AlertDialogSections slot) {
+    _placeChildInSlot(child, slot);
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, _AlertDialogSections slot) {
+    _placeChildInSlot(child, slot);
```

Comment:
i think you can assert false here. this will never be called.

Reply:
Done.